### PR TITLE
Binding Drawer and CodeEditor fixes

### PR DESCRIPTION
**Pull request by deanhannigan**
## Description
Some general refactoring and bug fixes for the Binding Drawer and CodeEditor components

## Addresses
- Fix for race condition for environment variables when loading the automations section
  - Environment variables load asynchronously to their own store and would frequently not be loaded in time.
  - This meant display issues for existing bindings and no entries for environment variables in the binding drawer.
- Fixed binding parsing for `Longform` text areas in row automation steps
  - Only the `runtimeBinding` value was visible, e.g. `{{ env.[testOpts] }}`
  - This now properly displays as `{{ env.testOpts }}`
  - Most bindings for automations don't have readable forms but any that do should display correctly now.
- Added general blur event to `CodeEditor`
  - As an inline field the `onChange` was far too chatty as it triggered an update request per keystroke.
  - Switched automation row steps to update JSON fields on blur, like other text areas.
- Added value reactivity to `CodeEditor`. Changing the base field content and opening the drawer would sometimes result in race conditions. This was apparent in a number of areas in the builder. 
  - This could have the effect of locking in the value for the binding drawer and potentially never refreshing the UI.
  - When the CodeEditor receives a new value, different its current value, it will now properly refresh the editor contents.
- Removed all props from the `DrawerBindableSlot` slot element as they were not valid.
- Changed bindable drawer fields in row automation steps to `forceModal` => `true`. 
  - There was an inconsistency with other fields in automations. It's also a much nicer UX.
- Fix for `CodeEditor` onDestroy behaviour. Calls the CodeMirror destroy function as intended
- Minor refactor for `CodeEditor`

## Launchcontrol
Bug fixes and refactoring for the Binding Drawer and CodeEditor components

### CODE REVIEW

**Review by aptkingston — COMMENTED** [review]
Few comments added. Lots of great fixes here, nice work @deanhannigan 👌

File: packages/builder/src/components/common/CodeEditor/CodeEditor.svelte [hunk 1]
```diff
@@ -58,6 +58,64 @@
...
+   * @param value {string} the editor value
+   */
+  const refresh = (value, initialised, mounted) => {
+    if (!initialised || !mounted) {
+      queuedRefresh = true
+      return
+    }
+
+    if (editor.state.doc.toString() !== value || queuedRefresh) {
+      editor.dispatch({
```
[the remark this answers is not in the document]
Comment (deanhannigan) [reply packages/builder/src/components/common/CodeEditor/CodeEditor.svelte:112]: I did a bunch of testing around that kind of scenario and didn't run into anything untoward. I can have another look and see if there's anything else I can do to mitigate.

[the remark this answers is not in the document]
Comment (aptkingston) [reply packages/builder/src/components/common/CodeEditor/CodeEditor.svelte:112]: Don't worry, that'll be fine then.

File: packages/builder/src/components/automation/SetupPanel/AutomationBlockSetup.svelte [hunk 2]
```diff
@@ -70,6 +70,7 @@
 
   // Stop unnecessary rendering
   const memoBlock = memo(block)
+  const memoEnvVariables = memo($environment.variables)
```
[the remark this answers is not in the document]
Comment (aptkingston) [reply packages/builder/src/components/automation/SetupPanel/AutomationBlockSetup.svelte:73]: We'll rewrite this after V3.

**Review by aptkingston — APPROVED** [review]

**Pull request merged**